src/main/java/frc5124/robot2020/commands/shooter/ShootFromMidTrench.java

```diff
@@ -0,0 +1,65 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc5124.robot2020.commands.shooter;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc5124.robot2020.RobotMap;
+import frc5124.robot2020.subsystems.Shooter;
+import frc5124.robot2020.subsystems.Loader;
+
+public class ShootFromMidTrench extends CommandBase {
+  private Shooter m_shooter;
+  private Loader m_loader;
+  
+  /**
+   * Creates a new setShootVelocity.
+   */
+  public ShootFromMidTrench (Shooter shooter, Loader loader) {
+    m_shooter = shooter;
+    m_loader = loader;
+    addRequirements(m_loader);
+    addRequirements(m_shooter);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_shooter.startShooter(RobotMap.ShooterMap.midTrenchShootRPM);
+    SmartDashboard.putBoolean("ShooterRunning", true);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() { 
+    // m_shooter.currentWatch(RobotMap.ShooterMap.lineShootRPM);
+    if (m_shooter.getVelocity() >= RobotMap.ShooterMap.midTrenchShootRPM-20 && m_loader.getAppliedOutput() == 0) {
```

Comment:
this line still only checks if we are below the shooting threshold, but does not consider if we are spiking above the threshold

Reply:
I suggest using the logic math.abs(getVelocity()-shootRPM) <= 20 to capture the range above and below the set point

Reply:
I see this logic is copied in other places. consider adding a method in the shooter subsystem that can give a boolean response when we have a valid firing solution